### DISCUSSION TITLE
In VSCode language extension while comment toggling, comment indicator goes to 1 column left side from cursor pointer

### DISCUSSION THREAD

**Discussion by jitendra19** [microsoft/vscode-discussions#2363] [Extension Development QnA] [answered]
I am creating a language extension, we don't use 1st column in this language purposely. So My language code start from 2nd column only. Now when I comment using _"CTRL+/"_ (comment toggling) language code statement, _Comment indicator (!)_ goes to first column (which is invalid in my language), my code shift by 1 column to right side.

I have to explicitly move comment indicator to 2nd column. then only comments are recognised by Textmate grammar. Please checkout the below image.

![image](https://github.com/user-attachments/assets/4c73e288-3e93-4d50-aa66-5c6893a2b145)

Is there a way to configure such thing in vscode language extension.

**Comment by alefragnani** [comment]
Hi @jitendra19 ,

I would suggest you to add that _extra column_ directly in `lineComment` entry in your `yourLanguage.configuration.json` file:

Something like this

```json
{
    "comments": {
        "lineComment": " !",
        "blockComment": [ "{", "}" ]
    },
    ...
}
```

And while using the `Toggle Line Comment` command, VS Code will respect your configuration and the `!` will be added in the second column.

But, a quick test on one of my language extensions, removing the line comment didn't work, and instead, another comment was added. And if the code is indented, sometimes the comment was added in the second column _after the indent_.

So, I'm not it is a starting point for you, but if you don't find another alternative, maybe worth a try.

Hope this helps

**Comment by SwatiMohapatra** [comment]
@alefragnani Hey , I tried the above solution and the toggle stopped working. Any suggestions how to fix this further?

**Reply by alefragnani** [reply, marked as answer]
Unfortunately not. As I commented, once I used a whitespace as the first character, the  `Toggle Line Comment` stopped _toggling_. I tried updating the `.tmLanguage` file contained on my language extension, but didn't make any difference.

I'm not familiar with this area of VS Code source, but it seems the _internals_ of the `Toggle Line Comment` command does ignore leading whitespace while deciding if it should remove comments or not (function `_analyzeLines` in `LineCommentCommand` class). If that's the case, I suppose you should ask for a feature request, so the VS Code team could evaluate if/how this could be achieved.

Hope this helps

**Reply by aiday-mar** [reply]
Hi thanks for this comment. So if I understood correctly are you saying that if the comment does not start with a leading whitespace you can toggle (add and remove comments). But if your comment does start with a leading whitespace, then the toggling does not work? This sounds like a bug, and I invite you to file an issue for this.

**Reply by RedCMD** [reply]
~~@aiday-mar you've said `not` twice by mistake~~
> But if your comment does not start with a leading whitespace, then the toggling does not work?

I can confirm
defining the comment token as '`#`' it does toggle
but when defined as '` #`' it does **not** toggle
resulting in ` #  #  #  #  # ` (when pressing ctrl+/ five times)

**Reply by aiday-mar** [reply]
Okay I will file an issue about this behavior.

**Comment by RedCMD** [comment]
there is a issue and PR for this
* https://github.com/microsoft/vscode/issues/234464
* https://github.com/microsoft/vscode/pull/243283#issuecomment-2857730744

@aiday-mar

**Reply by aiday-mar** [reply]
Hi @RedCMD thanks for the ping. So the PR above always adds the comment at the beginning of the line independently of the indentation. Is this is the desired behavior @jitendra19 ? Do you always want to add the comment at the beginning of the line on column 2?
